app/src/main/java/org/ieeeguc/ieeeguc/models/Committee.java

```diff
@@ -93,4 +96,54 @@ public void onResponse(Call call, Response response){
         });
 
     }
+
+    public static void addCommittee(String token,String name,String description,final HTTPResponse HTTP_RESPONSE){
+        OkHttpClient client = new OkHttpClient();
+        JSONObject j = new JSONObject();
+        try {
+            j.put("name", name);
+            j.put("description",description);
+            RequestBody body = RequestBody.create(CONTENT_TYPE,j.toString());
+            Request request = new Request.Builder()
+                    .url("http://ieeeguc.org/api/committee")
+                    .header("user_agent","Android")
+                    .header("Authorization",token)
+                    .post(body)
+                    .build();
+            client.newCall(request).enqueue(new Callback() {
+                @Override
+                public void onFailure(Call call, IOException e) {
+                    HTTP_RESPONSE.onFailure(-1, null);
+                    call.cancel();
+                }
+
+                @Override
+                public void onResponse(Call call, Response response) throws IOException {
+
+                    int code = response.code();
+                    String body = response.body().string();
+
+                    try {
+                        JSONObject j = new JSONObject(body);
+                        if(code/100 == 2)
+                        {
+                            HTTP_RESPONSE.onSuccess(code,j);
+                        }
+                        else
+                        {
+                            HTTP_RESPONSE.onFailure(code,j);
+                        }
+                    } catch (JSONException e) {
+                        HTTP_RESPONSE.onFailure(500,null);
+                    }
+
+                    response.close();
+                }
+            });
+        } catch (JSONException e) {
+
+            HTTP_RESPONSE.onFailure(-1,null);
```

Comment:
if the error is in the building of the body then the error is not in the connection thus the status code is not -1
However, the status code should be 500 as it is a problem from the application itself